op-batcher/batcher/channel_builder.go

```diff
@@ -398,46 +404,51 @@ func (c *ChannelBuilder) Close() {
 }
 
 // TotalFrames returns the total number of frames that were created in this channel so far.
-// It does not decrease when the frames queue is being emptied.
 func (c *ChannelBuilder) TotalFrames() int {
 	return c.numFrames
 }
 
 // HasFrame returns whether there's any available frame. If true, it can be
-// popped using NextFrame().
+// dequeued using NextFrame().
 //
 // Call OutputFrames before to create new frames from the channel out
 // compression pipeline.
 func (c *ChannelBuilder) HasFrame() bool {
```

Comment:
Not a strong opinion, IMO `HasPendingFrame()` feels more intuitive & accurate.

Reply:
I went with your suggestion in the end.